play/src/containers/GameInProgressContainer.tsx

```diff
@@ -1,118 +1,60 @@
+  const handleRetry = () => {
+    setErrorText({title1: 'Trying to reconnect...', title2: ''});
+    setRetry(retry + 1);
   };
+  
+  // loads game data from local storage
+  // if no game data, redirects to splashscreen
+  const pregameModel = useFetchLocalData();
+  // uses local game data to subscribe to gameSession 
+  // fetches gameSession first, then subscribes to data, finally returns object with loading, error and gamesession 
+  const  subscription = useFetchAndSubscribeGameSession(pregameModel.gameSessionId, apiClient, retry);
 
-  switch (currentState) {
-    case GameSessionState.CHOOSE_CORRECT_ANSWER:
-      return isPregameCountdown ? (
-        <PregameCountdown
-          handlePregameTimerFinished={handlePregameTimerFinished}
-        />
-      ) : (
-        <GameInProgress
-          {...gameSession}
-          teamAvatar={teamAvatar}
-          answerChoices={answerChoices}
-          teamId={teamId}
-          score={score}
-          addTeamAnswerToTeamMember={addTeamAnswerToTeamMember}
-        />
-      );
-    case GameSessionState.CHOOSE_TRICKIEST_ANSWER:
-    case GameSessionState.PHASE_1_DISCUSS:
-    case GameSessionState.PHASE_2_DISCUSS:
+  // this listens for changes to the retry state and updates the error text accordingly
+  React.useEffect(() => {
+    setErrorText({title1: t('joingame.errormodal.title1'), title2: subscription.error ?? 'Game Session Could Not Be Found'});
+  }, [retry, subscription.error, t]);
+  
+  // if gamesession is loading/errored/waiting for teacher to start game
+  if (!subscription.gameSession){
+    // if errored, show howToPlay page and error modal
+    if (subscription.error){
       return (
-        <GameInProgress
-          {...gameSession}
-          teamAvatar={teamAvatar}
-          answerChoices={answerChoices}
-          teamId={teamId}
-          score={score}
-          addTeamAnswerToTeamMember={addTeamAnswerToTeamMember}
-        />
-      );
-    case GameSessionState.PHASE_1_RESULTS:
-    case GameSessionState.PHASE_2_RESULTS:
-      return (
-        <PhaseResults
-          {...gameSession}
-          gameSession={gameSession}
-          currentQuestionIndex={gameSession!.currentQuestionIndex ?? 0} // eslint-disable-line @typescript-eslint/no-non-null-assertion
-          teamAvatar={teamAvatar}
-          teamId={teamId}
-          answerChoices={answerChoices}
-          score={score}
-          handleUpdateScore={handleUpdateScore}
-        />
-      );
-    case GameSessionState.PHASE_2_START:
-      return <StartPhase2 />;
-    case GameSessionState.FINAL_RESULTS:
-    default:
-      return (
-        <FinalResults
-          {...gameSession}
-          currentState={currentState}
-          score={score}
-          selectedAvatar={teamAvatar}
-          teamId={teamId}
-          leader={leader}
-        />
+        <>
+          <HowToPlay isLoading={false} isError />
+          <AlertModal errorText={errorText} retry={retry} handleRetry={handleRetry}/>
```

Comment:
I think you mentioned there might be an alternate way to do this (without passing down the booleans as props into the components themselves). Is there a cleaner approach than this?

Reply:
cleaned up the props here so it's just a single `mode` enum that changes based on the received status from the custom hook